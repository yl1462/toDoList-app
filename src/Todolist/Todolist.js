import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import config from '../config'

class Todolist extends Component {

  handleDelete = (id) => {
    fetch(`${config.API_ENDPOINT}/api/todo/${id}`, {
        method:'DELETE',
        headers:{
            'Content-Type':'application/json'
        }
    })
    
    .then(() => {
      this.props.deleteToDo(id)
      this.props.history.push('/home')
    })
    .catch(err => {
        this.setState({err})
    })
}

  render() {
    return (

      <div className="toDoList">

        {/* add new to do button on top of the page */}
        <button><Link to='/add'>Add New</Link></button>

        {/* each to do item shows the title and description */}
        <ul>
          {
            this.props.todos.map((todo, index) => (
              <li key={index}>
                <p className="todotitle">
                  {todo.title}<br />
                </p>
                <p>
                  {/* let user edit or delete their to do items */}
                  {todo.description}<br />
                  <button><Link to={{ pathname: `/edit/${todo.id}`, state: { todo, index } }}>Edit</Link> </button> <button onClick={() => this.handleDelete(todo.id)}>Delete</button>
                </p>
              </li>
            ))
          }
        </ul>
      </div>

    );
  }
}

export default Todolist;
