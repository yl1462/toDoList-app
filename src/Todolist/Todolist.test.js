import React from 'react';
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Todolist from './Todolist'

describe('Component does not crash', () => {
  //test if page loads
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Todolist todos={[]}/>
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe(`Todolist component`, () => {
  //test if test data loads into each to do item
  const props = {
    todos: [
      {
        "title": "walk the dog",
        "description": "Dogs"
      },

      {
        "title": "walk the cat",
        "description": "Cats"        
      },

      {
        "title": "walk the pig",
        "description": "Pigs"        
      },

      {
        "title": "walk the bird",
        "description": "Birds"
      },
    ],
    history: { push: () => {} },
    match: { params: "" }
  }

  it('renders a todo in ul for each todos in array', () => {
    const ul = shallow(<Todolist {...props} />)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
