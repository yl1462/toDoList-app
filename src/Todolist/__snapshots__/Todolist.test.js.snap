// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todolist component renders a todo in ul for each todos in array 1`] = `
<ul>
  <li
    key="0"
  >
    <p
      className="todotitle"
    >
      walk the dog
      <br />
    </p>
    <p>
      Dogs
      <br />
      <button>
        <Link
          to={
            Object {
              "pathname": "/edit/undefined",
              "state": Object {
                "index": 0,
                "todo": Object {
                  "description": "Dogs",
                  "title": "walk the dog",
                },
              },
            }
          }
        >
          Edit
        </Link>
         
      </button>
       
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </p>
  </li>
  <li
    key="1"
  >
    <p
      className="todotitle"
    >
      walk the cat
      <br />
    </p>
    <p>
      Cats
      <br />
      <button>
        <Link
          to={
            Object {
              "pathname": "/edit/undefined",
              "state": Object {
                "index": 1,
                "todo": Object {
                  "description": "Cats",
                  "title": "walk the cat",
                },
              },
            }
          }
        >
          Edit
        </Link>
         
      </button>
       
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </p>
  </li>
  <li
    key="2"
  >
    <p
      className="todotitle"
    >
      walk the pig
      <br />
    </p>
    <p>
      Pigs
      <br />
      <button>
        <Link
          to={
            Object {
              "pathname": "/edit/undefined",
              "state": Object {
                "index": 2,
                "todo": Object {
                  "description": "Pigs",
                  "title": "walk the pig",
                },
              },
            }
          }
        >
          Edit
        </Link>
         
      </button>
       
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </p>
  </li>
  <li
    key="3"
  >
    <p
      className="todotitle"
    >
      walk the bird
      <br />
    </p>
    <p>
      Birds
      <br />
      <button>
        <Link
          to={
            Object {
              "pathname": "/edit/undefined",
              "state": Object {
                "index": 3,
                "todo": Object {
                  "description": "Birds",
                  "title": "walk the bird",
                },
              },
            }
          }
        >
          Edit
        </Link>
         
      </button>
       
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </p>
  </li>
</ul>
`;
